From: Michael Froman <mfroman@mozilla.com>
Date: Thu, 10 Oct 2024 13:42:00 +0000
Subject: Bug 1921707 - add filtering for new build directory location in
 third_party/libwebrtc/webrtc.gni r=ng,webrtc-reviewers

Differential Revision: https://phabricator.services.mozilla.com/D224545
Mercurial Revision: https://hg.mozilla.org/mozilla-central/rev/f18c05287ce831369f44715a4174facff4f12652
---
 webrtc.gni | 14 ++++++++++++--
 1 file changed, 12 insertions(+), 2 deletions(-)

diff --git a/webrtc.gni b/webrtc.gni
index 60cb81271c..2efd99096f 100644
--- a/webrtc.gni
+++ b/webrtc.gni
@@ -536,7 +536,7 @@ template("rtc_test") {
       min_sdk_version = 21
       target_sdk_version = 23
       deps += [
-        "//build/android/gtest_apk:native_test_instrumentation_test_runner_java",
+        "//chromium/build/android/gtest_apk:native_test_instrumentation_test_runner_java",
         webrtc_root + "sdk/android:native_test_jni_onload",
         webrtc_root + "sdk/android:base_java",
         webrtc_root + "test:native_test_java",
@@ -984,11 +984,21 @@ template("rtc_library") {
       modified_deps = []
       foreach (dep, deps) {
         newdep = string_replace(dep, "//third_party/", "//libwebrtc/third_party/")
+        newdep = string_replace(newdep, "//build", "//chromium/build")
         modified_deps += [ newdep ]
       }
       deps = []
       deps = modified_deps
     }
+    if (build_with_mozilla && defined(data_deps)) {
+      modified_deps = []
+      foreach (dep, data_deps) {
+        newdep = string_replace(dep, "//build/", "//chromium/build/")
+        modified_deps += [ newdep ]
+      }
+      data_deps = []
+      data_deps = modified_deps
+    }
   }
 }
 
@@ -1026,7 +1036,7 @@ template("rtc_executable") {
     if (is_win) {
       deps += [
         # Give executables the default manifest on Windows (a no-op elsewhere).
-        "//build/win:default_exe_manifest",
+        "//chromium/build/win:default_exe_manifest",
       ]
     }
 
